ArjunaJTS/jts/classes/com/arjuna/ats/internal/jts/context/ContextManager.java

```diff
@@ -296,7 +296,7 @@ public final void purgeActions (String threadId)
 
     public final void purgeActions ()
     {
-        purgeActions(ThreadUtil.getThreadId());
+        purgeActions(ThreadUtil.removeThreadId(Thread.currentThread()));
```

Comment:
If Thread.currentThread() is not in the THREAD_ID map then this statement will be equivalent to
```java
purgeActions(null)
```
and this will call
```java
 public final ControlWrapper popAction (String threadId)
```
which blows up when it calls
```java
Object arg = otsCurrent.get(threadId);
```
because otsCurrent is a Hashtable which 
```java
@throws NullPointerException if the specified key is null
```